tf_quant_finance/black_scholes/brownian_bridge.py

```diff
@@ -0,0 +1,182 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Methods for brownian bridges.
+
+These can be used in Monte-Carlo simulation for payoff with continuous barrier.
+Indeed, the Monte-Carlo simulation is inherently discrete in time, and to
+improve convergence (w.r.t. the number of time steps) for payoff with continuous
+barrier, adjustment with brownian bridge can be made.
+"""
+
+import tensorflow.compat.v2 as tf
+
+
+def brownian_bridge_double(*,
+                           x_start,
+                           x_end,
+                           variance,
+                           upper_barrier,
+                           lower_barrier,
+                           n_cutoff=3,
+                           dtype=None,
+                           name=None):
+  """ Probability of not touching any of the barriers
+  (`upper_barrier` and `lower_barrier`) for a Brownian Bridge starting at
+  `x_start`, ending at `x_end` and with variance `variance`, assuming that
```

Comment:
Just a nitpick but the first sentence in the docstring should be an imperative verb with the first line being <=80 symbols ending with a period and separated by a blank line. Say, here,  

"""Computes probabilities of not touching the boundaries  for a Brownian Bridge.

 For a Brownian Bridge starting at `x_start`, ending at `x_end` and with the specified `variance` ... blah blah